pkg/controller/direct/directbase/directbase_controller.go

```diff
@@ -318,9 +340,21 @@ func (r *reconcileContext) doReconcile(ctx context.Context, u *unstructured.Unst
 	if err != nil {
 		if unwrappedErr, ok := lifecyclehandler.CausedByUnresolvableDeps(err); ok {
 			logger.Info(unwrappedErr.Error(), "resource", k8s.GetNamespacedName(u))
-			return r.handleUnresolvableDeps(ctx, u, unwrappedErr)
+
+			// if we have failed to "FIND" the resource and that is caused by an error deemed to be about its unresolved
+			// dependencies BUT the object is being deleted right now AND its dependecies have been deleted, then we will
+			// never encounter a "ready" condition for its dependencies
```

Comment:
note I pulled out this change in its own PR too for visibility-- https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/4666

Reply:
@gemmahou FYI as this feels like part of the deletion logic improvement you're working on.

Reply:
wouldn't it make more sense for us to add these checks in directbase too? As opposed to having to add the per controller checks?